hotspot/make/linux/makefiles/gcc.make

```diff
@@ -212,7 +212,7 @@ ifeq ($(USE_CLANG), true)
   WARNINGS_ARE_ERRORS += -Wno-return-type -Wno-empty-body
 endif
 
-WARNING_FLAGS = -Wpointer-arith -Wsign-compare -Wundef -Wunused-function -Wunused-value -Wformat=2 -Wreturn-type
+WARNING_FLAGS = -Wpointer-arith -Wsign-compare -Wundef -Wunused-function -Wunused-value -Wformat=2 -Wreturn-type -Wno-stringop-overflow
```

Comment:
disabling the warning instead of fixing the code is not the proper way to deal with this

Reply:
Thank you for your code review. 
The same solution is used in later JDK versions. https://bugs.openjdk.org/browse/JDK-8320212 ; 
So JDK8 can also use this solution.

Reply:
No, rather the other way round: please fix that in later JDK versions as well.

I’m only the Debian packager for JDK 8, so I have no relationship to the other versions. This doesn’t make my critique invalid.

Reply:
thanks.  Ok, I'll try to fix it forward through the code;
Since I'm not sure how many errors there are yet, this may take a while. I'll try;

Reply:
Hi mirabilos，
I looked up some information; reference: https://gcc.gnu.org/bugzilla/show_bug.cgi? ID=106297. I also found that the gcc12 in my test environment was not a pure version but a version after secondary development, which affected the misjudgment of the jdk8 compilation error.
I built and tested using https://ftp.gnu.org/gnu/gcc/. The -Werror=stringop-overflow= error does not exist, and hotspot/src/share/vm/gc_implementation/g1/concurrentMark.cpp does not need to be modified.
The verification is as follows:
![image](https://github.com/openjdk/jdk8u-dev/assets/9154227/87ed51a0-ef3a-4e24-8b7f-c900446b247a)

Therefore, I think the changes in gcc.make and concurrentMark.cpp should be removed.
thanks;